doc/src/tutorial/compile_cmake.dox

```diff
@@ -33,18 +35,77 @@ After executing the above, you can either open the generated solution with Visua
+
+    build_path> cmake -G Ninja -DPA_BUILD_TESTS=ON /path/to/portaudio/source
+
+Then you would add `-DPA_BUILD_EXAMPLES=ON` like so:
+
+    build_path> cmake -G Ninja -DPA_BUILD_TESTS=ON **-DPA_BUILD_EXAMPLES=ON** /path/to/portaudio/source
+
+Afterwards, continue the build process as normal. Once the build finishes, there will be an `examples/` folder containing all the examples where you can try out PortAudio in action.
+
+@section Building Tests
+
+Unlike some of our other build systems (e.g., autotools), tests are not built automatically. In that case, you can add
+
+    -DPA_BUILD_TESTS=ON
+
+to your CMake build options. For example:
 
-PortAudio defines the following CMake targets:
+    build_path> cmake -G <Your Generator> -DPA_BUILD_TESTS=ON [other CMake options...] [/path/to/portaudio/source]
+
+Then follow the rest of the normal build process. After the build finishes, you will notice a `test/` folder where you can run any test to ensure PortAudio is working correctly.
+
+<b>Note</b>: PortAudio _does not_ support CTest at the moment. Contributions are welcome.
+
+@section cmake_build_options CMake Build Options
+
+PortAudio's CMake build system contains the following build options categorized below.
+
+Backend options (all default to `ON` if available except for the OSS, ASIO, and skeleton backends):
+
+- **`-DPA_USE_ALSA=ON|OFF`**: (Linux/Unix-like/etc. Only). Use the ASLA host API. Requires ALSA.
+    - **`-DPA_ALSA_DYNAMIC=ON|OFF`**: Dynamically load libasound with dlopen using PaAlsa_SetLibraryPathName.
+- **`-DPA_USE_ASIO=ON|OFF`**: (Windows Only). Use the ASIO host API. Requires the ASIO SDK[2].
+- **`-DPA_USE_AUDIOIO=ON|OFF`**: (Solaris Only). Use the audioio host API.
+- **`-DPA_USE_DS=ON|OFF`**: (Windows Only). Use the DirectSound host API.
+- **`-DPA_USE_JACK=ON|OFF`**: Use the JACK host API. Requires JACK.
+- **`-DPA_USE_SKELETON=ON|OFF`**: Use the skeleton host API (no audio output).
+- **`-DPA_USE_OSS=ON|OFF`**: (Linux/Unix-like/etc. Only). Use the OSS host API[1]. Requires OSS.
+- **`-DPA_USE_PULSEAUDIO=ON|OFF`**: (Linux/Unix-like/etc. Only). Use the PulseAudio host API. Requires PulseAudio to be present.
+- **`-DPA_USE_SNDIO=ON|OFF`**: (Linux/Unix-like/etc. Only). Use the sndio host API.
+- **`-DPA_USE_WASAPI=ON|OFF`**: (Windows Only). Use the WASAPI host API.
+- **`-DPA_USE_WDMKS=ON|OFF`**: (Windows Only). Use the WDMKS host API.
+    - **`-DPA_USE_WDMKS_DEVICE_INFO=ON|OFF`**: Use the WDMKS API for device info.
+- **`-DPA_USE_WMME=ON|OFF`**: (Windows Only). Use the MME host API.
+
+[1] The ASIO SDK is only available under a proprietary license. This license imposes many requirements including conditions and limitations on distributing binaries that you build. Be sure to read this page from Steinberg: https://www.steinberg.net/developers. Also be advised that enabling ASIO can cause an impact on initialization performance and reliability, so exercise caution when enabling this backend.
```

Comment:
Please add a link to #696, otherwise this is too vague and readers will not know what to make of this.